src/telemetrydeck-signal.test.tsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable import/no-unassigned-import */
+/* eslint-disable import/extensions */
+import React from "react";
+import { fireEvent, render, screen, act } from "@testing-library/react";
+import "cross-fetch/polyfill";
+import "./__mocks__/mock-global";
+import { setupServer } from "msw/node";
+import mockSignal from "./__mocks__/mock-signal";
+import Setup from "./test-utils/setup-td";
+import { handlers } from "./test-utils/handlers";
+import mockQueue from "./__mocks__/mock-queue";
+import mockFlush from "./__mocks__/mock-flush";
+import { createTelemetryDeck } from "./create-telemetrydeck";
+
+const server = setupServer(...handlers);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+const component = "dashboard";
+const path = "/dashboard";
+
+test("signals are sent when pressing the button", async () => {
+  const td = createTelemetryDeck({ appID: "123456789", clientUser: "anonymous" });
+
+  /**
+   * This event listener is used to count the number of requests sent to the server.
+   */
+  let requestCount = 0;
+  server.events.on("request:start", () => {
+    requestCount += 1;
+  });
```

Comment:
This code is duplicated in each test. That's bad practice. Try to either implement it in the beforeEach lifecycle hook or at least modularize it.

Reply:
As I see it we have different mocked requests in every tests. The second test just happens to have the same but apart from that, they are not the same.